runhouse/resources/hardware/cluster_factory.py

```diff
@@ -85,51 +79,27 @@ def cluster(
             "Cluster factory method can only accept one of `host` or `ips` as an argument."
         )
 
-    if name and all(
-        not x
-        for x in [
-            host,
-            ssh_creds,
-            server_port,
-            server_host,
-            server_connection_type,
-            ssl_keyfile,
-            ssl_certfile,
-            den_auth,
-            kwargs,
-        ]
-    ):
-        # If only the name is provided
-        return Cluster.from_name(name, dryrun)
-
-    if host and ("localhost" in host or ":" in host):
-        # If server_connection_type is not specified, we
-        # assume we can hit the server directly via HTTP
-        server_connection_type = server_connection_type or ServerConnectionType.NONE
-        if ":" in host:
-            # e.g. "localhost:23324" or <real_ip>:<custom port> (e.g. a port is already open to the server)
-            host, client_port = host.split(":")
-            kwargs["client_port"] = client_port
-
-    server_connection_type = server_connection_type or (
-        ServerConnectionType.TLS
-        if ssl_certfile or ssl_keyfile
-        else ServerConnectionType.SSH
-    )
-
-    if server_port is None:
-        if server_connection_type == ServerConnectionType.TLS:
-            server_port = DEFAULT_HTTPS_PORT
-        elif server_connection_type == ServerConnectionType.NONE:
-            server_port = DEFAULT_HTTP_PORT
-        else:
-            server_port = DEFAULT_SERVER_PORT
-
-    if name in RESERVED_SYSTEM_NAMES:
-        raise ValueError(
-            f"Cluster name {name} is a reserved name. Please use a different name which is not one of "
-            f"{RESERVED_SYSTEM_NAMES}."
+    if name:
+        alt_options = dict(
+            host=host,
+            ssh_creds=ssh_creds,
+            server_port=server_port,
+            server_host=server_host,
+            server_connection_type=server_connection_type,
+            ssl_keyfile=ssl_keyfile,
+            ssl_certfile=ssl_certfile,
+            den_auth=den_auth,
+            kwargs=kwargs,
         )
+        # Filter out None/default values
+        alt_options = {k: v for k, v in alt_options.items() if v is not None}
+        try:
+            c = Cluster.from_name(name, dryrun, alt_options=alt_options)
```

Comment:
we want to enable this `alt_options` behavior right? this is a changefrom before

Reply:
Y